app/models/piece.rb

```diff
@@ -8,4 +8,91 @@ class Piece < ActiveRecord::Base
   scope :rooks, -> { where(type: 'Rook') }
   scope :queens, -> { where(type: 'Queen') }
   scope :kings, -> { where(type: 'King') }
+
+  def is_obstructed?(row, column)
+    if moving_horizontally?(row)
+      is_horizontal_obstructed?(column)
+    elsif moving_vertically?(column)
+      is_vertical_obstructed?(row)
+    elsif moving_diagonally?(row, column)
+      is_diagonal_obstructed?(row, column)
+    else
+      fail 'Input is invalid!'
+    end
+  end
+
+  private
+
+  NONE = 0
+  UP = 1
+  DOWN = -1
+  RIGHT = 1
+  LEFT = -1
+
+  def moving_horizontally?(row)
+    self.row == row
+  end
+
+  def moving_vertically?(column)
+    self.column == column
+  end
+
+  def moving_diagonally?(row, column)
+    (self.row - row).abs == (self.column - column).abs
+  end
+
+  def moving_up_and_to_the_right?(row, column)
+    self.row < row && self.column < column
+  end
+
+  def moving_up_and_to_the_left?(row, column)
+    self.row < row && self.column > column
+  end
+
+  def moving_down_and_to_the_right?(row, column)
+    self.row > row && self.column < column
+  end
+
+  def is_horizontal_obstructed?(column)
```

Comment:
I think that erroring on `is_`.

Is new to Rubocop.

So let's follow what rubocop says and remove `is_`.

I think `horizontally_obstructed?(destination_column)` works nicely